### PR TITLE
Support indices preceding the base symbol

### PR DESCRIPTION
**Pull request by Leedehai**
Hi this PR adds native support for attaching indices preceding the base symbol ("prescripts", "left index"). This is very useful in denoting isotopes, etc. and mirrors the composition capabilities offered by Latex. For a demo, please see the modified test images.

The modified function signature is `attach(base, bottom, top, prebottom, pretop)`.

This PR fixes https://github.com/typst/typst/issues/690.

### CODE REVIEW

File: library/src/math/attach.rs [hunk 1]
```diff
@@ -19,33 +20,45 @@ pub struct AttachElem {
...
 
         let base = ctx.layout_fragment(&base)?;
 
+        macro_rules! take_attachment {
+            ($F: ident) => {
+                self.$F(ctx.styles())
+                    .map(|elem| ctx.layout_fragment(&elem))
+                    .transpose()?
+            };
+        }
```
Comment (laurmaedje) [inline library/src/math/attach.rs:52]: Could this be done with a closure instead of a macro?

**Review by laurmaedje — COMMENTED** [review]
This looks good, I have just one thing I'm unsure about: The naming with `top`, `bottom`, `prebottom` and `pretop`. We could also consider `top-left`, `top-right`, `bottom-left`, and `bottom-right`. Or we could have six properties: The four `top-left`, ... as well as `top` and `bottom`. The latter two would smartly fit into the right spots (right or limit). Maybe its a bad idea, but the current names don't feel nice to me.

File: library/src/math/attach.rs (hunk 1 above)
Comment (Leedehai) [reply library/src/math/attach.rs:52]: Yup. New patch coming!

**Comment by Leedehai** [comment]
Whoops. I messed up the commit diff on GitHub. Closing this PR in favor of the cleaner #825.

**Pull request closed without merging**